### PR TITLE
Tms/merchant revenue #13

### LINKED ISSUE #13: Merchant revenue
GET /api/v1/merchants/:id/revenue

### LINKED ISSUE #14: Merchant revenue by date
GET /api/v1/merchants/:id/revenue?date=x

### LINKED ISSUE #20: Merchant revenue by date
GET /api/v1/merchants/revenue?date=x

### PR DESCRIPTION
**Pull request by tmikeschu**
Worked with @Dpalazzari - ready for review

- Adds endpoint for merchant revenue
- Adds model instance methods for those endpoints

* Room to potentially refactor the Merchant#revenue... methods with a scope/lambda..

Needs in-depth review. All tests passing.

Closes #13 
Closes #20 
Closes #14 
(14 and 20 are duplicates)

@sallymacnicholas

### CODE REVIEW

**Comment by Dpalazzari** [comment]
@tmikeschu Is this just the method we worked on with Toni, right?

**Comment by tmikeschu** [comment]
@Dpalazzari yes, plus the other spec expectation for passing a date query parameter

**Comment by tmikeschu** [comment]
Thanks @ToniRib !

**Comment by ToniRib** [comment]
Looks like you guys put this together pretty well!

**Comment by Dpalazzari** [comment]
@tmikeschu  Does this revenue method convert cents into dollars? Does it pass Spec harness?

**Comment by tmikeschu** [comment]
@Dpalazzari good catch. I'll add the dollars string conversion and check the spec harness.

File: spec/requests/api/v1/merchants/revenue_intelligence_spec.rb [hunk 1]
```diff
@@ -0,0 +1,40 @@
...
+        invoice.invoice_items << create(:invoice_item, unit_price: 150, quantity: 2)
+      end
+      new_date = merchant.invoices.first.created_at + 10.days
+      merchant.invoices.first.update(created_at: new_date)
+      merchant.reload
+
+      get "/api/v1/merchants/#{merchant.id}/revenue?date=#{new_date}"
+
+      expect(response).to be_success
+      expect(response.body).to eq('400')
```
Comment (Dpalazzari) [inline spec/requests/api/v1/merchants/revenue_intelligence_spec.rb]: @tmikeschu Should we stick to Toni's convention is make sure the (response.body).to include('400')?

Comment (tmikeschu) [reply spec/requests/api/v1/merchants/revenue_intelligence_spec.rb]: @Dpalazzari already on this!

**Comment by tmikeschu** [comment]
@Dpalazzari let's wait until we have serializers going so we can format the revenue values

**Comment by tmikeschu** [comment]
Worked with @Dpalazzari to polish off this PR. Ready for merge.

- Added a custom serializer for formatting revenue.
- Adjusted formatting for invoice item price

Needs in-depth review, all tests passing AND endpoints and relationships in the spec harness are all passing, along with the two analytics methods.

**Pull request merged**